staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3645,7 +3645,18 @@ type NodeSpec struct {
 // NodeConfigSource specifies a source of node configuration. Exactly one subfield (excluding metadata) must be non-nil.
 type NodeConfigSource struct {
 	metav1.TypeMeta `json:",inline"`
```

Comment:
`metav1.TypeMeta` doesn't make sense in a non-top-level object... we should drop this

Reply:
These are serialized to disk to record references to the assigned and last-known-good configs.

Reply:
See e.g. `pkg/kubelet/kubeletconfig/checkpoint/store.SetCurrent`, `pkg/kubelet/kubeletconfig/checkpoint/store.writeRemoteConfigSource`, `pkg/kubelet/kubeletconfig/checkpoint.remoteConfigMap.Encode`

Reply:
hmm... it's odd to expose fields in our REST API that aren't allowed to be set (or aren't honored if they are)

Reply:
I guess we could wrap it in a new type called e.g. `SerializedNodeConfigSource` instead. Not sure what tombstoning the currently inlined TypeMeta would look like though....